src/main/kotlin/org/jitsi/nlj/transform/node/incoming/IncomingStatisticsTracker.kt

```diff
@@ -75,23 +100,59 @@ class IncomingStatisticsTracker : ObserverNode("Incoming statistics tracker") {
         val parentStats = super.getNodeStats()
         return NodeStatsBlock(name).apply {
             addAll(parentStats)
-            val stats = getCurrentStats()
-            stats.forEach { ssrc, streamStats ->
+            val stats = getSnapshot()
+            stats.ssrcStats.forEach { ssrc, streamStats ->
                 addStat("source: $ssrc")
-                addStat(streamStats.getSnapshot().toString())
+                addStat(streamStats.toString())
             }
         }
     }
+
+    fun getSnapshot(): IncomingStatisticsSnapshot {
+        val now = System.currentTimeMillis()
+        return IncomingStatisticsSnapshot(
+            bitrate.getRate(now),
+            packetRate.getRate(now),
+            bytesReceived,
+            packetsReceived,
+            ssrcStats.map { (ssrc, stats) ->
+                Pair(ssrc, stats.getSnapshot())
+            }.toMap()
+        )
+    }
 }
 
+class IncomingStatisticsSnapshot(
```

Comment:
so this one is per receiver? `ReceiverIncomingStatisticsSnapshot`?

Reply:
Isn't that superfluous?

Reply:
I was thinking that `IncomingStatisticsSnapshot` is still ambiguous as to the scope of the stats (per stream or per receiver), so including `Receiver` suggests it's across the entire receiver?

Reply:
When I read `ReceiverIncomingStatisticsSnapshot` I don't immediately understand what the `Receiver` part stands for (in fact I thought you wanted to me add it to differentiate from the sender), so I prefer to keep the name shorter. I think the standard usage is clear enough:
```
transceiverStats.getIncomingStats().getBitrate();
```

Reply:
ok, fair enough.